### ISSUE TITLE
[BUG] Not operator doesn't work on range key

### ISSUE THREAD

**Issue by akowalsk** [dynamoose/dynamoose#1535]
### Summary:

After upgrading from dynamoose 2.8.3 to 3.1.0, a query started producing a ValidationException error (ValidationException: Invalid operator used in KeyConditionExpression: <>).  This can be reproduced with the following code:

### Code sample:
```
const dynamoose = require('dynamoose');
const { Schema } = dynamoose;

dynamoose.aws.ddb.local();

const ExampleSchema = new Schema(
  {
    hash_key: {
      type: String,
      hashKey: true,
    },
    range_key: {
      type: String,
      rangeKey: true,
    },
    other_attribute: {
      type: Boolean,
    },
  },
  { timestamps: true }
);

dynamoose.model('example', ExampleSchema);

async function query() {
  (await dynamoose.logger()).providers.set(console);
  const Example = dynamoose.model('example');
  const res = await Example.query('hash_key')
    .eq('hk-1')
    .where('range_key')
    .not()
    .eq('rk-1')
    .filter('other_attribute')
    .eq(false)
    .exec();
  return res;
}

query().then(console.log).catch(console.log);
```
### Current output and behavior (including stack trace):

aws:dynamodb:describeTable:request - {
    "TableName": "example"
}
aws:dynamodb:describeTable:response - {
    "$metadata": {
        "httpStatusCode": 200,
        "requestId": "48d814e3-2c41-4896-a19e-bd0f32d30004",
        "attempts": 1,
        "totalRetryDelay": 0
    },
    "Table": {
        "AttributeDefinitions": [
            {
                "AttributeName": "hash_key",
                "AttributeType": "S"
            },
            {
                "AttributeName": "range_key",
                "AttributeType": "S"
            }
        ],
        "CreationDateTime": "2022-10-16T19:39:25.836Z",
        "ItemCount": 0,
        "KeySchema": [
            {
                "AttributeName": "hash_key",
                "KeyType": "HASH"
            },
            {
                "AttributeName": "range_key",
                "KeyType": "RANGE"
            }
        ],
        "ProvisionedThroughput": {
            "LastDecreaseDateTime": "1970-01-01T00:00:00.000Z",
            "LastIncreaseDateTime": "1970-01-01T00:00:00.000Z",
            "NumberOfDecreasesToday": 0,
            "ReadCapacityUnits": 1,
            "WriteCapacityUnits": 1
        },
        "TableArn": "arn:aws:dynamodb:ddblocal:000000000000:table/example",
        "TableName": "example",
        "TableSizeBytes": 0,
        "TableStatus": "ACTIVE"
    }
}
aws:dynamodb:query:request - {
    "FilterExpression": "#a2 = :v2",
    "ExpressionAttributeNames": {
        "#a2": "other_attribute",
        "#qha": "hash_key",
        "#qra": "range_key"
    },
    "ExpressionAttributeValues": {
        ":v2": {
            "BOOL": false
        },
        ":qhv": {
            "S": "hk-1"
        },
        ":qrv": {
            "S": "rk-1"
        }
    },
    "TableName": "example",
    "KeyConditionExpression": "#qha = :qhv AND #qra <> :qrv"
}
ValidationException: Invalid operator used in KeyConditionExpression: <>
    at throwDefaultError (...../node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:8:22)
    at deserializeAws_json1_0QueryCommandError (...../node_modules/@aws-sdk/client-dynamodb/dist-cjs/protocols/Aws_json1_0.js:2140:51)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ...../node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
    at async ...../node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:13:20
    at async StandardRetryStrategy.retry (...../node_modules/@aws-sdk/middleware-retry/dist-cjs/StandardRetryStrategy.js:51:46)
    at async ...../node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:6:22
    at async main (...../node_modules/dynamoose/dist/aws/ddb/internal.js:6:20)
    at async ...../node_modules/dynamoose/dist/ItemRetriever.js:105:32
    at async query (...../dynamooseBug.js:28:15) {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    requestId: '1c2f01e3-89fe-433b-92e1-111e4b104699',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  __type: 'com.amazon.coral.validate#ValidationException'
}

### Expected output and behavior:

Based on the example in the Condition documentation, I expected the query to run and produce the result.  I also tried using the "filter" syntax instead of "where" (i.e. `Example.query('hash_key').eq('hk-1').filter('range_key').not().eq('rk-1').filter('other_attribute').eq(false).exec()`) but it produced the same error and query expression.  

### Environment:

Operating System: macOS
Operating System Version: 12.5
Node.js version (`node -v`): 16.15.1
NPM version: (`npm -v`): 8.11.0
Dynamoose version: 3.1.0

### Other information (if applicable):

"@aws-sdk/client-dynamodb": "^3.165.0"

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by jdukewich** [comment]
Unfortunately, this is a limitation of DynamoDB. This isn't valid DynamoDB behavior. When querying a global secondary index, you must use an equality condition on the hash key, and the range key can only be one of a certain few operators or the `begins_with` function.

[DynamoDB query docs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html#Query.KeyConditionExpressions) list which operators you can use with the range key when querying.

Can you try applying the not equals on the filtering step rather than the initial query?

**Comment by akowalsk** [comment]
If by "applying the not equals on the filtering step" you mean running the query like this:

`Example.query('hash_key').eq('hk-1').filter('range_key').not().eq('rk-1').filter('other_attribute').eq(false).exec()`

then I did try that.  Dynamoose converts that back to the same query expression and I get the error.  That is, It still puts "range_key" in the KeyConditionExpression instead of in the FilterExpression, even though I'm using "filter" instead of "where".

Is there a different query syntax that you had in mind?

**Comment by jdukewich** [comment]
Yes that is what I had in mind, but I didn't realize that would still be converted. I guess Dynamoose figures that since `range_key` is indeed a range key, it will always force that into the KeyConditionExpression. I think for now you'd have to do this filter yourself application side, after Dynamoose stuff happens. 

Looking at it now, I'm wondering if this is considered a bug or just a missing feature (i.e. being able to use attributes that are part of the key in the filter expression). I'll defer that to the core maintainers

**Comment by akowalsk** [comment]
I believe the filter syntax "worked" in 2.8.x and started to error when I upgraded to 3.1.  I'm not actually sure if it did filter out the range key successfully in 2.8, but it at least didn't error.  I used the "where" syntax in the example because it was the simplest way to get the error, my original application was using the "filter" syntax.

I'm already filtering it out on the application side, as you suggested.  The main reason I thought it would count as a bug is that it worked previously.  Now that I look carefully at the documentation, it does [indicate that any rangeKey will be used in a KeyConditionExpression](https://dynamoosejs.com/guide/Query#modelqueryfilter).   I had always thought "where" was specifically reserved for rangeKeys and "filter" was for anything, but that could be from the 1.x days.
